### DISCUSSION TITLE
Documentation Update Proposal: Accessing Trivy Operator Metrics through a Grafana Dashboard

### DISCUSSION THREAD

**Discussion by 02fa** [aquasecurity/trivy-operator#997] [Help & Support]
I'm proposing to update [docs / tutorials / grafana-dashboard.md](https://github.com/aquasecurity/trivy-operator/compare/main...Wang-W-e-i:trivy-operator:to-match-examples-file-names?expand=1) for readability and consistency of examples.

In addition, with changes in service/trivy-operator exposure (ClusterIP vs. LoadBalancer), the tutorial fails at me, starting from line [137](https://github.com/aquasecurity/trivy-operator/blob/957eb9b443582d9929e811e724a54de8ef61f417/docs/tutorials/grafana-dashboard.md?plain=1#L137) `kubectl port-forward service/trivy-operator -n trivy-system 5000:80`.

My challenge could be that my cluster doesn't have any LoadBalancer objects ATM.

Q1: How the tutorial would look like with the new way of exposing service/trivy-operator? Shall it be updated to include the requirement to have a valid LoadBalancer object or something?

Q2: It could be related to Q1, but Prometheus cannot discover serviceMonitor, created by trivy-operator. I can read metrics, produced by trivy-operator's end point (`kubectl get ep -ntrivy-system`), but nothing else. How could I make Prometheus aware of these metrics, please?

**Comment by chen-keinan** [comment]
@AnaisUrlichs fyi

**Reply by 02fa** [reply]
Watched recent @AnaisUrlichs video on YouTube, quite impressive! My challenge is I am not a listening type of person. The way I consume information is by reading - it's much faster.

I found a few reports in un-related projects that `serviceMonitorSelectorNilUsesHelmValues` needs to be set to `false` in order to pick up `serviceMonitor`s from other name spaces. Changed it to `false` on `prometheus-values.yaml`, and upgraded `kube-prometheus-stack` Helm chart. Prometheus has happily picked up serviceMonitor/trivy-system/trivy-operator/0 (1/1 up).

Absolutely smashing, Aqua Security! Thank you for maintaining this project!

If that matters, my cluster is kubeadm with external etcd.

```
kubectl version --short
Client Version: v1.26.1
Kustomize Version: v4.5.7
Server Version: v1.26.1
```

```
helm version --short
v3.11.1+g293b50c
```

```
helm list -A
NAME          	NAMESPACE   	REVISION	UPDATED                                	STATUS  	CHART                       	APP VERSION
prom          	monitoring  	2       	2023-02-22 16:13:28.612574816 +0000 UTC	deployed	kube-prometheus-stack-45.1.1	v0.63.0    
trivy-operator	trivy-system	1       	2023-02-20 16:10:48.581564831 +0000 UTC	deployed	trivy-operator-0.11.1       	0.11.1
```

**Comment by AnaisUrlichs** [comment]
@Wang-W-e-i thank you for the details and the feedback! We try to add tutorials to the docs that are representative of longer YouTube tutorials -- did you see this one on Prometheus and Grafana with Trivy? https://aquasecurity.github.io/trivy-operator/v0.11.1/tutorials/grafana-dashboard/

Let us know if there is anything in the docs that you think we can/should improve 🙂
